packages/flutter_tools/lib/src/runner/flutter_command_runner.dart

```diff
@@ -506,7 +506,10 @@ class FlutterCommandRunner extends CommandRunner<void> {
     final List<String> projectPaths = fs.directory(rootPath)
       .listSync(followLinks: false)
       .expand((FileSystemEntity entity) {
-        return entity is Directory ? _gatherProjectPaths(entity.path) : <String>[];
+        if (entity is Directory && !entity.path.contains('.dart_tool')) {
```

Comment:
Actually, I'd use `path.split` and make sure that there is a component equal to ".dart_tool" in the list, so that if someone creates a directory with a suffix of "foo.dart_tool", it will still get looked at.  (Yeah, a nit to be sure.)
```suggestion
        if (entity is Directory && !path.split(entity.path).contains('.dart_tool')) {
```

Reply:
Good idea!